rules/throw-new-error.js

```diff
@@ -9,13 +9,14 @@ const errorTypes = [
 	'TypeError',
 	'URIError'
 ];
+const customerError = /^(?:[A-Z][a-z0-9]*)*Error$/;
 
 const create = context => ({
 	ThrowStatement: node => {
 		const arg = node.argument;
 		const error = arg.callee;
 
-		if (arg.type === 'CallExpression' && errorTypes.indexOf(error.name) !== -1) {
+		if (arg.type === 'CallExpression' && (errorTypes.indexOf(error.name) !== -1 || customerError.test(error.name))) {
```

Comment:
If the regex is well made, we don't need the `errorTypes.indexOf(error.name) !== -1` check anymore. If you remove it and all tests pass, it should be fine to remove.